### PR TITLE
fix(vue): autosuggest prop not working correctly

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
- Issue Type - Bug

- Notion Link - https://www.notion.so/appbase/Searchbox-Fix-autosuggest-property-159e39a529004485b286845fc4ac3a22

- Description - Passing false to the autosuggest prop was still triggering the default query and showing the suggestions list.

Fixes -
-> Don't trigger defaultQuery when autosuggest is false.
-> Don't fetch recentSearches when autosuggest is false.
-> Don't display suggestions when autosuggest is false.
-> When autosuggest is false then results should get updated on each key stroke.
-> Debounce should work as expected.

Demo -
https://www.loom.com/share/23b335965af249d782290c5e9e9bcbd3

### CODE REVIEW

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Check the comments on react-searchbox PR.

**Comment by anik-ghosh-au7** [comment]
> @anik-ghosh-au7 Check the comments on react-searchbox PR.

@bietkul made the changes - https://www.loom.com/share/005709fda79249f78c17c67f999d938d

File: packages/vue-searchbox/src/components/SearchComponent.jsx [hunk 1]
```diff
@@ -58,7 +58,8 @@ const SearchComponent = {
 		render: VueTypes.func,
 		// subscribe on changes,
 		subscribeTo: VueTypes.arrayOf(VueTypes.string),
-		triggerQueryOnInit: VueTypes.bool.def(true)
+		triggerQueryOnInit: VueTypes.bool.def(true),
+		autosuggest: VueTypes.bool.def(true)
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchComponent.jsx]: @anik-ghosh-au7 Remove `autosuggest`

File: packages/vue-searchbox/src/components/SearchComponent.jsx [hunk 2]
```diff
@@ -224,9 +225,9 @@ const SearchComponent = {
 		}, subscribeTo);
 	},
 	mounted() {
-		const { triggerQueryOnInit } = this.$props;
+		const { triggerQueryOnInit, autosuggest } = this.$props;
 		const componentInstance = this.getComponentInstance();
-		if (triggerQueryOnInit) {
+		if (triggerQueryOnInit && autosuggest) {
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchComponent.jsx]: @anik-ghosh-au7 Remove `autosuggest`

File: packages/vue-searchbox/src/components/URLParamsProvider.jsx [hunk 3]
```diff
@@ -6,16 +6,17 @@ const URLParamsProvider = {
 	inject: ['searchbase'],
 	props: {
 		id: VueTypes.string.isRequired,
+		autosuggest: VueTypes.bool.def(true)
 	},
 	mounted() {
-		const { id } = this.$props;
+		const { id, autosuggest } = this.$props;
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/URLParamsProvider.jsx]: Fix it

**Review by bietkul — CHANGES_REQUESTED** [review]

**Review by bietkul — APPROVED** [review]

**Pull request merged**